addons/mail/static/tests/qunit_suite_tests/components/call_main_view_tests.js

```diff
@@ -0,0 +1,57 @@
+/** @odoo-module **/
+
+import {
+    start,
+    startServer,
+} from '@mail/../tests/helpers/test_utils';
+
+QUnit.module('mail', {}, function () {
+QUnit.module('components', {}, function () {
+QUnit.module('call_main_view_tests.js');
+
+QUnit.test('Join and leave a call', async function (assert) {
+    assert.expect(5);
+
+    const pyEnv = await startServer();
+    const mailChannelId = pyEnv['mail.channel'].create({});
+    const { click, openDiscuss } = await start({
+        discuss: {
+            params: {
+                default_active_id: `mail.channel_${mailChannelId}`,
+            },
+        },
+    });
+    await openDiscuss();
+    await click('.o_ThreadViewTopbar_callButton');
+
+    assert.containsOnce(
+        document.body,
+        '.o_CallView',
+        "Should have a call view"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_CallParticipantCard',
+        "Should have a call participant card"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_CallMainView_controls',
+        "Should have call controls"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ThreadViewTopbar_callButton',
+        "Should not have a join call button anymore"
+    );
+    await click('.o_CallActionList_callToggle');
+    assert.containsNone(
+        document.body,
+        '.o_CallView',
+        "Should not have a call view"
+    );
```

Comment:
The assertion comment do not understand the functional importance of these assertions. Why there should be a call view?

I know it's because you click on call button. Please keep in mind the feedback of assertion comments failing is on QUnit UI: when you see a test _"Join and leave a call"_, the exact scenario is not super clear (it's named like a tour).

If it crashes on _"Should have a call view"_, I should guess it's because you were attempting to make a call, and the concept "call view" should have some trace on UI. It's not a guess when you look in the code, but remember that QUnit UI lack crucial things like the `click()`. Poor assertion comments waste debug time.

This applies for all comments. Prefer making a small test that makes a single assertion (that way the test is named very specifically and failure gives immediate understand what fails precisely).